src/receive_imf.rs

```diff
@@ -728,12 +728,20 @@ async fn add_parts(
                     }
                 }
 
-                // The next block checks if the message was sent with verified encryption
-                // and sets the protection of the 1:1 chat accordingly.
-                if is_partial_download.is_none()
+                // Check if the message was sent with verified encryption and set the protection of
+                // the 1:1 chat accordingly.
+                let chat = match is_partial_download.is_none()
                     && mime_parser.get_header(HeaderDef::SecureJoin).is_none()
                     && !is_mdn
+                    && context
+                        .get_config_bool(Config::VerifiedOneOnOneChats)
```

Comment:
Also moved this check here. If verified chats are not enabled, no need to load the chat from the db

Reply:
Moving the check does change semantics, though:

The current behavior is documented in this comment:

```rust
    /// Feature flag for verified 1:1 chats; the UI should set it
    /// to 1 if it supports verified 1:1 chats.
    /// Regardless of this setting, `chat.is_protected()` returns true while the key is verified,
    /// and when the key changes, an info message is posted into the chat.
    /// 0=Nothing else happens when the key changes.
    /// 1=After the key changed, `can_send()` returns false and `is_protection_broken()` returns true
    /// until `chat_id.accept()` is called.
```

With this PR, 1:1 chats aren't be shown as verified in the UI anymore at all. While we could do this, this would need some discussion, update the documentation, and make sure that verified 1:1 chats can still be downgraded to unverified if `VerifiedOneOnOneChats` is not set.

Reply:
Maybe the config should have been called something else than `VerifiedOneOnOneChats`, since this name suggests that when disabling it, 1:1 chats will be shown as unverified, while in reality, disabling it just "disarms" them, for UIs that aren't adapted to verified 1:1 chats. Anyway, it's not that important, since we the plan is to remove it after all the UIs are adapted, anyway.

Reply:
Thanks, got it. Moved the check back. So, now i only don't set chat protection to `ProtectionBroken` if it's already set. I think it is ok to keep the current name for the setting, moreover if we remove it soon.